src/core/lv_obj_tree.c

```diff
@@ -416,6 +416,127 @@ uint32_t lv_obj_get_child_count_by_type(const lv_obj_t * obj, const lv_obj_class
     return cnt;
 }
 
+#if LV_USE_OBJ_NAME
+
+void lv_obj_set_name(lv_obj_t * obj, const char * name)
+{
+    LV_ASSERT_NULL(obj);
+    lv_obj_allocate_spec_attr(obj);
+
+    if(!obj->spec_attr->name_static && obj->spec_attr->name) lv_free((void *)obj->spec_attr->name);
+
+    if(name == NULL) {
+        obj->spec_attr->name = NULL;
+        obj->spec_attr->name_static = 1;
+    }
+    else {
+        obj->spec_attr->name = lv_strdup(name);
+        obj->spec_attr->name_static = 0;
+    }
+}
+
+void lv_obj_set_name_static(lv_obj_t * obj, const char * name)
+{
+    LV_ASSERT_NULL(obj);
+    lv_obj_allocate_spec_attr(obj);
+
+    if(!obj->spec_attr->name_static && obj->spec_attr->name) lv_free((void *)obj->spec_attr->name);
+
+    obj->spec_attr->name = name;
+    obj->spec_attr->name_static = 1;
+}
+
+const void * lv_obj_get_name(lv_obj_t * obj)
+{
+    LV_ASSERT_NULL(obj);
+    if(obj->spec_attr == NULL) return NULL;
+    else return obj->spec_attr->name;
+}
+
+void lv_obj_get_name_resolved(lv_obj_t * obj, char buf[], size_t buf_size)
+{
+    LV_ASSERT_NULL(obj);
+    if(obj->spec_attr && obj->spec_attr->name) {
+        lv_strlcpy(buf, obj->spec_attr->name, buf_size);
+    }
+    /*Craft a name if not set. E.g. "lv_button_1"*/
+    else {
+        uint32_t idx = lv_obj_get_index_by_type(obj, obj->class_p);
+        lv_snprintf(buf, buf_size, "%s_%d", obj->class_p->name, idx + 1);
+    }
+}
+
+lv_obj_t * lv_obj_get_child_by_name(const lv_obj_t * parent, const char * path)
+{
+
+    if(parent->spec_attr == NULL || path == NULL) return NULL;
+
+    while(*path) {
+        const char * segment = path;
+        uint32_t len = 0;
+
+        /* Calculate the length of the current segment */
+        while(path[len] && path[len] != '/')
+            len++;
+
+        bool found = false;
+        lv_obj_t * child = NULL;
+        /* Look for a child whose resolved name exactly matches the segment */
+        for(uint32_t i = 0; i < parent->spec_attr->child_cnt; i++) {
```

Comment:
```suggestion
        const uint16_t child_cnt = lv_obj_get_child_count(parent);
        for(uint16_t i = 0; i < child_cnt; i++) {
```

1. If a getter function exists isn't it better to use it in case the data structure changes in the future ? 
2. Just noticed that the function `uint32_t lv_obj_get_child_count(const lv_obj_t * obj)` should be changed to  `uint16_t lv_obj_get_child_count(const lv_obj_t * obj)` as `child_cnt` is defined as a `uint16_t`
3. Maybe this loop can be extracted to a different function ? There's alot of repetition with code in lines 520..526

```c
static lv_obj_t * lv_find_child(lv_obj_t * parent, const char * name, size_t len) {
    const uint16_t child_cnt = lv_obj_get_child_count(parent);
    for(uint16_t i = 0; i < child_cnt; i++) {
        lv_obj_t * child = parent->spec_attr->children[i];
        char child_name_resolved[128]; 
        lv_obj_get_name_resolved(child, child_name_resolved, sizeof(child_name_resolved));
        if(lv_strncmp(child_name_resolved, name, len) == 0 &&
            child_name_resolved[len] == '\0') {
            return child;
        }
    }
    return NULL;
}
```

Reply:
> If a getter function exists isn't it better to use it in case the data structure changes in the future ?

I have used it directly for performance reasons, but here it shouldn't matter much.

> Just noticed that the function uint32_t lv_obj_get_child_count(const lv_obj_t * obj) should be changed to uint16_t lv_obj_get_child_count(const lv_obj_t * obj) as child_cnt is defined as a uint16_t


For simplicity we are targeting to use `(u)int32_t`  where possible. 

> Maybe this loop can be extracted to a different function ? There's alot of repetition with code in lines 520..526


Good idea, done!